### PR TITLE
Update the approach of checking the CPUArchitecture

### LINKED ISSUE #522: The issue of getCPUArchitecture
**Describe the bug**
Current cpu_model doesn't support the latest Intel Sapphire Rapids CPU. When I wanted to add it, I found kepler use [**archspec**,](https://github.com/archspec/archspec) a python package to retrieve the cpu information.

So I read the source code of archspec project, it uses cpu flags to judge the CPU Architecture, which is not very accurate and always need to maintain that repo manually for the new CPU.

I think it's better to use **cpuid** to get the correct CPU Architecture.

kepler_node_nodeInfo{cpu_architecture="unknown"}

**Expected behavior**
kepler_node_nodeInfo{cpu_architecture="Sapphire Rapids"}

### PR DESCRIPTION
**Pull request by LeiZhou-97**
fix #522 

1. Use cpuid instead of archspec to retrieve correct CPUArchitecture information. 
2. Add Sapphire Rapids cpu model.

Signed-off-by: Zhou,Lei <lei.zhou@intel.com>

### CODE REVIEW

**Comment by rootfs** [comment]
thank you for the `cpuid` idea @LeiZhou-97!

File: pkg/collector/metric/node_metric.go [hunk 1]
```diff
@@ -106,7 +123,7 @@ func getCPUArchitecture() (string, error) {
 		if err := dec.Decode(&p); err == io.EOF {
 			break
 		}
-		if strings.HasPrefix(myCPUModel, p.Name) {
+		if strings.Contains(myCPUModel, p.Architecture) {
```
Comment (rootfs) [inline pkg/collector/metric/node_metric.go:126]: if we can get uniform cpu architecture names, we don't need to read `normalized_cpu_arch.csv` any more; the file is there to help resolve different naming conventions (temporary workaround)

File: pkg/collector/metric/node_metric.go [hunk 2]
```diff
@@ -85,11 +84,29 @@ func getCPUArchitecture() (string, error) {
 		klog.V(2).Infof("cpu arch override: %v\n", cpuArchOverride)
 		return cpuArchOverride, nil
 	}
-	output, err := exec.Command("archspec", "cpu").Output()
+
+	// use cpuid to get CPUArchitecture
+	grep := exec.Command("grep", "uarch")
+	output := exec.Command("cpuid", "-1")
```
Comment (rootfs) [inline pkg/collector/metric/node_metric.go:90]: per `cpuid` [manpage](https://man7.org/linux/man-pages/man4/cpuid.4.html), it is a x86 tool.  Can you refactor the code to externalize the `getCPUArchitecture` call in separate files and conditionally compile it to use `cpuid` on x86 and `archspec` on arm64?

Comment (LeiZhou-97) [reply pkg/collector/metric/node_metric.go]: Updated. Move all unrelated node_metric func to utils.go and distinguish x86/arm64. (if you think it's better to create a new file for cpu architecture, I can edit it again).

**Review by rootfs — APPROVED** [review]

**Pull request merged**